core/src/main/scala/org/bykn/bosatsu/Matchless.scala

```diff
@@ -810,63 +930,139 @@ object Matchless {
         LetMut(anon, rest)
       }
 
+    def staticMatch(arg: Expr, branches: NonEmptyList[
+        (Pattern[(PackageName, Constructor), Type], Expr)
+      ]
+    ): Option[Expr] =
+      arg match {
+        case Literal(Lit.Str(s)) =>
```

Comment:
We could extend this very easily to any Literal which is probably all we need to handle here.

That said, maybe this code should really be pushed into TypedExprNormalization since applying these optimizations could open other normalizations.